lib/analytics/course_students_csv_builder.rb

```diff
@@ -5,6 +5,9 @@
 class CourseStudentsCsvBuilder
   def initialize(course)
     @course = course
+    @editors = @course.articles_courses.pluck(:user_ids).flatten
+    new_articles = @course.articles_courses.where(new_article: 1)
+    @new_articles_revisions = Revision.where(article_id: new_articles)
```

Comment:
This is likely to be a very expensive query on a production server, which will have millions of Revisions and possibly tens or hundreds of thousands of new_article revisions.

Reply:
Noted, I'll try and find another way..

Reply:
What if, I limit the scope of the query to also include new articles, for instance 
@new_articles_revisions = Revision.where(article_id: new_articles, new_article: 1)

That way, it does not retrieve all the revisions of an article, just the first one for each article..